ui/src/App.js

```diff
@@ -22,37 +23,57 @@ class App extends Component {
       console.log('Connecting to ' + port.device_.productName + '...');
       port.connect().then(() => {
         console.log('Connected to port:', port);
-        port.onReceive = data => {
-          let textDecoder = new TextDecoder();
-          console.log(textDecoder.decode(data));
-        }
-        port.onReceiveError = error => {
-          console.log('Receive error: ' + error);
-        };
+        port.onReceive = this.handleSerialMessage.bind(this)
+        port.onReceiveError = this.handleSerialError.bind(this)
+
+        // Try to load our bitcoin address
+        let textEncoder = new TextEncoder();
```

Comment:
This is becoming "callback hell". We'll need to refactor soon.

Reply:
I think we can include it in some kind of "device" class with isConnected(), connect(), requestMasterPublicKey() etc methods. The easiest way to get rid of the callback hell right now is to switch to async/await syntax.